site/snippets/enhanced-apis-example/viem-client/nft/example.ts

```diff
@@ -0,0 +1,18 @@
+import { AlchemyProvider } from "@alchemy/aa-alchemy";
+import { sepolia } from "viem/chains";
+import { createAlchemyEnhancedApiClient } from "./factory.js";
+
+const provider = new AlchemyProvider({
+  chain: sepolia,
+  apiKey: "YOUR_API_KEY",
+});
+
+export const client = createAlchemyEnhancedApiClient({
+  chain: sepolia,
+  rpcUrl: `${sepolia.rpcUrls.alchemy.http[0]}/YOUR_API_KEY`,
+});
```

Comment:
Also, missed bringing this up from previous PR:

can we NOT create a new function `createAlchemyEnhancedApiClient` but just add an optional parameter to `createPublicErc4337Client` parameters?

For cleaner Dev ex

Reply:
yes for token, not for nft. that said, it's unnecessary to use `createPublicErc4337Client` in this example since we're just calling the enhanced apis, not the aa apis, via this client.

the current `createPublicErc4337Client` requires an http transport that works for 1193 json rpc, not restful apis. the nft viem client needs a custom transport to handle the restful apis.

we'd need to refactor `createPublicErc4337Client` to permit this. I have a branch up that does this to some degree for uo sim, but as Moldy mentioned live at one point - think we want to push users to suggestions 1 and 2 of using alchemy-sdk to extend the alchemy aa provider.

Reply:
Note my comment above. To simplify the dev ex, why not just list `alchemy-sdk-js` as not peer dep, but regular dep, and initialize Alchemy SDK for the users if using enhanced api option.

Maybe I am missing something with my understanding.

Reply:
We wanted to avoid dependency bloat because alchemy-sdk-js is built on `ethers.js`. Since e-apis are not required to use aa-sdk except for this add-on, we left it as a peer dependency. I still think that's the right move.